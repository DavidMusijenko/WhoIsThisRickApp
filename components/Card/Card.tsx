import { StyleSheet } from "react-native";
import { Text, View } from "../Themed";
import { Image } from "react-native";

const Card = ({ character }: any) => {
  return (
    <View>
      <Image
        style={{ width: 350, height: 350 }}
        source={{
          uri: character?.image,
        }}
      />
      <Text>{character?.name}</Text>
      <Text>{character?.status}</Text>
      <Text>{character?.species}</Text>
      <Text>Last known location: {character?.location.name}</Text>
    </View>
  );
};

export default Card;

const styles = StyleSheet.create({});
