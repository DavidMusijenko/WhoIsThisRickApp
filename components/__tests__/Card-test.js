import renderer from 'react-test-renderer';
import Card from '../Card/Card';

//write a test for the Card component
//it should render correctly
//it should match the snapshot
//it should have a title prop
//it should have a description prop
//it should have a image prop

it(`renders correctly`, () => {
  const tree = renderer.create(<Card />).toJSON();

  expect(tree).toMatchSnapshot();
});
