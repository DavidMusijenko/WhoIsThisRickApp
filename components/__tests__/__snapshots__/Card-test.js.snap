// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has a description prop 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "height": 750,
        "width": 750,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 40,
          "padding": 30,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 24,
          "padding": 0,
        },
      ]
    }
  >
     
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#eee",
        },
        Object {
          "height": 1,
          "marginVertical": 30,
          "width": "80%",
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 24,
          "padding": 10,
        },
      ]
    }
  >
     Last known location: 
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 24,
          "padding": 10,
        },
      ]
    }
  />
</View>
`;

exports[`has a image prop 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "height": 750,
        "width": 750,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 40,
          "padding": 30,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 24,
          "padding": 0,
        },
      ]
    }
  >
     
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#eee",
        },
        Object {
          "height": 1,
          "marginVertical": 30,
          "width": "80%",
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 24,
          "padding": 10,
        },
      ]
    }
  >
     Last known location: 
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 24,
          "padding": 10,
        },
      ]
    }
  />
</View>
`;

exports[`has a title prop 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "height": 750,
        "width": 750,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 40,
          "padding": 30,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 24,
          "padding": 0,
        },
      ]
    }
  >
     
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#eee",
        },
        Object {
          "height": 1,
          "marginVertical": 30,
          "width": "80%",
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 24,
          "padding": 10,
        },
      ]
    }
  >
     Last known location: 
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 24,
          "padding": 10,
        },
      ]
    }
  />
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "height": 750,
        "width": 750,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 40,
          "padding": 30,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 24,
          "padding": 0,
        },
      ]
    }
  >
     
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#eee",
        },
        Object {
          "height": 1,
          "marginVertical": 30,
          "width": "80%",
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 24,
          "padding": 10,
        },
      ]
    }
  >
     Last known location: 
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 24,
          "padding": 10,
        },
      ]
    }
  />
</View>
`;
